src/integration_tests/datasets/dataset_endpoints_integration_test.py

```diff
@@ -0,0 +1,135 @@
+import pytest
+from unittest import TestCase
+from src.app.config.config_factory import config
+from src.test_data.dataset_test_data import dataset_metadata_collection_endpoints
+from src.test_data.dataset_test_data import dataset_unit_data_collection_endpoints
+from repositories.firebase.firebase_loader import firebase_loader
+from src.integration_tests.helpers.integration_helpers import (
+    cleanup,
+    generate_headers,
+    setup_session,
+    inject_wait_time,
+)
+from src.integration_tests.helpers.firestore_helpers import upload_dataset
+from google.cloud import firestore
+import logging
+
+class DatasetEndpointsIntegrationTest(TestCase):
+    """
+    Integration tests for the Dataset Endpoints.
+
+    This test covers uploading datasets, fetching metdata and unit data from firestore,
```

Comment:
'uploading dataset' isnt tested functionality. The tests only cover calling endpoints and ensuring the data received is correct